### DISCUSSION TITLE
Integration with Keycloak

### DISCUSSION THREAD

**Discussion by Nicknameee** [apache/camel-karavan#1327] [Q&A]
### Ask a question

Good afternoon, this discussion regards Keycloak integration as authentication provider for Apache Camel Karavan UI
We are interested in next features:
1) Logout - currently UI itself does not provide any identification information about currently authorized user, same thing with logout(no ui support for this)
We implemented Keycloak as authentication provider
![image](https://github.com/apache/camel-karavan/assets/98988322/21768b6a-54e1-4386-a5ab-b714b728596a)
Is it possible to configure communication of Apache Camel Karavan and Keycloak in order to get desired functionality - logout icon of some sort, etc
2) Information about current logged in user - username at least
It would be great if it is possible to identify who currently using the dashboard
If this feature not implemented - will it be, if not, is it possible to add it as contribution (get current source UI version, edit for our purposes and use it in future)
ghcr.io/apache/camel-karavan:4.5.1
Best regards

### Variant

Web Application

### Container Management (if applicable)

Kubernetes

### Operating System (if applicable)

None

### Version

4.5.0

**Comment by mgubaidullin** [comment]
If OIDC is properly configured https://github.com/apache/camel-karavan/blob/4d89d9ed441edb4ea4cccabb614a535a8613717e/karavan-app/src/main/resources/application.properties#L74

then Current User information and Logout Button are at the bottom of left menu panel

<img width="477" alt="Screenshot 2024-06-26 at 10 26 08 AM" src="https://github.com/apache/camel-karavan/assets/1379213/486cdc7e-80aa-4904-bbda-7fecd9e798c1">
